RecoEgamma/EgammaTools/src/EnergyScaleCorrection.cc

```diff
@@ -0,0 +1,431 @@
+      file.get();
+      continue;
+    }
+    
+    if(file.peek() == 35) { // 35 = #
+      file.ignore(1000, 10); // ignore the rest of the line until \n
+      continue;
+    }
+    
+    if(smearingType_ == UNKNOWN) { // trying to guess: not recommended
+      throw cms::Exception("ConfigError") <<"unknown smearing type";
+      
+    }else if(smearingType_ == GLOBE) {
+      file >> category >> unused >> etaMin >> etaMax >> r9Min >> r9Max >> runMin >> runMax >>
+	eMean >> errEMean >>
+	rho >> errRho >> phi >> errPhi;
+      
+      addSmearing(category, runMin, runMax, rho,  errRho, phi, errPhi, eMean, errEMean);
+      
+    } else if(smearingType_ == ECALELF) {
+      file >> category >> 
+	eMean >> errEMean >>
+	rho >> errRho >> phiString >> errPhiString;
+      
+      if(phiString=="M_PI_2") phi=M_PI_2;
+      else phi = std::stod(phiString);
+      
+      if(errPhiString=="M_PI_2") errPhi=M_PI_2;
+      else errPhi = std::stod(errPhiString);
+      
+      addSmearing(category, runMin, runMax, rho,  errRho, phi, errPhi, eMean, errEMean);
+      
+    } else {
+      file >> category >> rho >> phi;
+      errRho = errPhi = eMean = errEMean = 0;
+      addSmearing(category, runMin, runMax, rho,  errRho, phi, errPhi, eMean, errEMean);
+    }
+
+  }
+  
+  file.close();
+  return;
+}
+
+std::ostream& EnergyScaleCorrection::ScaleCorrection::print(std::ostream& os)const
+{
+  os <<  "( "<< scale_ << " +/- " << scaleErrStat_ << " +/- " << scaleErrSyst_ << " +/- " << scaleErrGain_ <<")" ;
+  return os; 
+}
+
+float EnergyScaleCorrection::ScaleCorrection::scaleErr(const std::bitset<kErrNrBits>& uncBitMask)const
+{
+  double totErr(0);
+  auto pow2 = [](const double& x){return x*x;};
+  
+  if(uncBitMask.test(kErrStatBitNr)) totErr+=pow2(scaleErrStat_);
+  if(uncBitMask.test(kErrSystBitNr)) totErr+=pow2(scaleErrSyst_);  
+  if(uncBitMask.test(kErrGainBitNr)) totErr+=pow2(scaleErrGain_);
+  
+  return std::sqrt(totErr);
+}
+
+std::ostream& EnergyScaleCorrection::SmearCorrection::print(std::ostream& os)const
+{
+  os << rho_ << " +/- " << rhoErr_ 
+     <<  "\t"
+     << phi_ << " +/- " << phiErr_
+     <<  "\t"
+     << eMean_ << " +/- " << eMeanErr_;
+  return os; 
+}
+
+//here be dragons
+//this function is nasty and needs to be replaced
+EnergyScaleCorrection::CorrectionCategory::CorrectionCategory(const std::string& category,int runnrMin,int runnrMax):
+  runMin_(runnrMin),runMax_(runnrMax),etaMin_(0),etaMax_(3),
+  r9Min_(-1),r9Max_(999),etMin_(0),etMax_(9999999),gain_(0)
+{
+  size_t p1, p2; // boundary
+
+  // eta region
+  p1 = category.find("absEta_");
+  if(category.find("absEta_0_1") != std::string::npos) {
+    etaMin_ = 0;
+    etaMax_ = 1;
+  } else if(category.find("absEta_1_1.4442") != std::string::npos) {
+    etaMin_ = 1;
+    etaMax_ = 1.479;
+  }
+  else if(category.find("absEta_1.566_2") != std::string::npos) {
+    etaMin_ = 1.479;
+    etaMax_ = 2;
+  }
+  else if(category.find("absEta_2_2.5") != std::string::npos) {
```

Comment:
is it so difficult to change the file content to contain what it means?

Reply:
While I 100% agree with you that this is awful (@shervin86 is the author and so I'm redirecting the complaints  to him),  changing this function will incur significant delay and remove any chance of this making the miniAOD. Its one of the few functions I didnt touch at all. 

A better approach would be a nice conddb object or at the very least a json /xml file.